### PR TITLE
Cache decorator implemented

### PR DESCRIPTION
**Pull request by thorium-zero**

### CODE REVIEW

**Review by Oleg-s-s — APPROVED** [review]
Nice)

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+    cache_vault = {}
+
+    def inner(*args) -> Any:
+        if args in cache_vault:
+            print("Getting from cache")
+            return cache_vault[args]
+        print("Calculating new result")
+        result = func(*args)
+        cache_vault[args] = result
+        return result
```
Comment (Oleg-s-s) [inline app/main.py:14]: You are returning the same value, try using only one return.

**Review by Oleg-s-s — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 1 above)
Comment (thorium-zero) [reply app/main.py]: Updated...

**Review by Oleg-s-s — APPROVED** [review]
Nice :+1:

**Pull request closed without merging**